.pre-commit-config.yaml

```diff
@@ -0,0 +1,13 @@
+repos:
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.6.5
+    hooks:
+      # Fix imports
+      - id: ruff
+        args: ["--fix", "--select", "I"]
+      # Linter (with fix)
+      - id: ruff
+        args: ["--fix"]
+      # Formatter
+      - id: ruff-format
+        types_or: ["python"]
```

Comment:
Do we want to let it run on Jupyter notebooks or not?

Reply:
I will actually say let's not. 
We often have jupyter notebooks that are meant to be filled in with code, which from the perspective of ruff will make them `broken`, which could prevent commits. People could of course use `--no-verify` if we want to try and format notebooks.